spec/models/driver_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+describe Driver do
+  describe 'driver validations' do
+    subject { driver.valid? }
+
+    let(:driver) do
+      build(
+        :driver,
+        name: name,
+        license_number: license_number,
+        mobile_number: mobile_number,
+        email_address: email_address,
+        age: age,
+        status: status
+      )
+    end
+    let(:name) { 'some name' }
+    let(:license_number) { 'license_number' }
+    let(:mobile_number) { '98765432' }
+    let(:email_address) { 'email@email.com' }
+    let(:age) { 18 }
+    let(:status) { true }
+
+    it { is_expected.to be_truthy }
+
+    context 'mobile number' do
+      context 'non singapore number' do
+        let(:mobile_number) { '0412345678' }
+
+        it { is_expected.to be_falsey }
+      end
+
+      context 'singapore number' do
+        let(:mobile_number) { '98765432' }
+
+        it { is_expected.to be_truthy }
+      end
+    end
+
+    context 'email address' do
+      context 'is not valid' do
+        let(:email_address) { '1.2' }
+
+        it { is_expected.to be_falsey }
+      end
+
+      context 'is valid' do
+        let(:email_address) { '1@2.3' }
+
+        it { is_expected.to be_truthy }
+      end
+    end
+  end
+
+  describe 'driver validations with shoulda matchers' do
```

Comment:
in future lets not put shoulda matchers and manual validations in separate blocks. since we wont ever do that in actual practice

Reply:
in that case, should i take it out of the block?

Reply:
sure lets combine them now then

Reply:
done